lib/pbench/client/oidc_admin.py

```diff
@@ -0,0 +1,148 @@
+from http import HTTPStatus
+import json
+
+import requests
+
+from pbench.server.auth import Connection
+
+
+class OIDCAdmin(Connection):
+    OIDC_REALM = "pbench-server"
+    OIDC_CLIENT = "pbench-dashboard"
+    ADMIN_USERNAME = "admin"
+    ADMIN_PASSWORD = "123"
```

Comment:
While `OIDC_REALM` is used multiple times within this class, when you create the admin account you use a literal username and the "canonical standard" admin password of `"admin"` which makes the `ADMIN_PASSWORD` definition here not only useless but wrong. 

In any case, I didn't spot a use of any of these except the `OIDC_REALM`. Maybe I missed them; but should these be removed?

This also ties into the fact that while `load_keycloak.sh` *defaults* the admin username and password to `admin`/`admin`, it allows them to be overridden, in which case this code (both the constants here and the literals in `get_admin_token`) are broken. If we allow the setup to be overridden (and we need to, as, e.g., the RDU2 Keycloak presumably *won't* be `admin`/`admin`), we need to be able to configure the functional tests as well in order to run against the staging (or production) deployment.

Reply:
> In any case, I didn't spot a use of any of these except the `OIDC_REALM`. Maybe I missed them; but should these be removed?

Yes, I should remove them.


> This also ties into the fact that while `load_keycloak.sh` _defaults_ the admin username and password to `admin`/`admin`, it allows them to be overridden, in which case this code (both the constants here and the literals in `get_admin_token`) are broken. If we allow the setup to be overridden (and we need to, as, e.g., the RDU2 Keycloak presumably _won't_ be `admin`/`admin`), we need to be able to configure the functional tests as well in order to run against the staging (or production) deployment.

Yeah, Instead of hardcoding the `ADMIN_USERNAME` and `ADMIN_PASSWORD`, I should first check the env variable for these values and default it to `admin` and `admin` respectively. keycloak.sh script does the same, it first checks the env variable of the same name and then defaults it to the admin/admin. This will allow us to run functional tests in different kinds of environments like staging/production.